core/capabilities/ccip/ccipevm/gas_helpers.go

```diff
@@ -52,15 +53,33 @@ func bytesForMsgTokens(numTokens int) int {
 
 // CalculateMessageMaxGas computes the maximum gas overhead for a message.
 func (gp EstimateProvider) CalculateMessageMaxGas(msg cciptypes.Message) uint64 {
+	maxGas, err := gp.CalculateMessageMaxGasWithError(msg)
+	if err != nil {
+		panic(err)
+	}
+	return maxGas
+}
+
+// CalculateMessageMaxGasWithError computes the maximum gas overhead for a message.
+// TODO: Add destGasOverhead, see:
+// https://github.com/smartcontractkit/chainlink/blob/23452266132228234312947660374fb393e96f1a/contracts/src/v0.8/ccip/FeeQuoter.sol#L97
+func (gp EstimateProvider) CalculateMessageMaxGasWithError(msg cciptypes.Message) (uint64, error) {
```

Comment:
`CalculateMessageMaxGasWithError` naming implies to me that we are doing something useful with the error. Panic'ing is ...useful, just not the usual application. I see both API's are public and exportable - is there a plan to deprecate `CalculateMessageMaxGas`